go/vt/wrangler/keyspace.go

```diff
@@ -459,13 +459,13 @@ func (wr *Wrangler) MigrateServedTypes(ctx context.Context, keyspace, shard stri
 	wr.Logger().Infof("WaitForDrain: Sleeping finished. Shutting down queryservice on old tablets now.")
 
 	rec := concurrency.AllErrorRecorder{}
+	// Refresh both source and destinations to make sure migrated primaries don't
+	// show up as primaries AND serving.
 	refreshShards := sourceShards
-	if reverse {
-		// For a backwards migration, we should refresh (disable) destination shards instead.
-		refreshShards = destinationShards
```

Comment:
We should check with @deepthi  and @sougou  about this. According with what we have discussed so far, this makes sense to me. But before this code actually used to refresh the source shards during a forward migration and destination shards during a reverse migration. 

So the original intent of this code was a bit different. I want to triple check that we are not missing anything here.

Reply:
To me, it actually makes more sense to refresh destinationShards during forward migration and sourceShards during reverse migration than what was being done earlier.
In order to cover all failover cases, I think what you are doing is better. Refresh both source and destination.
@sougou WDYT?

Reply:
We ran into different issues for both cases during a forward migration:
* In the destination shards, a failover left the shard in a state where it couldn't start QueryService because it was part of a split and that it should be running [vreplication](https://github.com/tinyspeck/vitess/blob/master/go/vt/vttablet/tabletmanager/tm_state.go#L265). This is the main bug that kicked off this discussion. 
* The opposite happens in the source shard: if a failover occurs right after the cutover, the source shard comes back in a state where it thinks the primary should be serving. Luckily this is more benign because the topology is correct and nothing is sending traffic to it.

I think we should cover both cases.

Reply:
That makes sense to me.